lib/sacastats_web.ex

```diff
@@ -78,6 +78,36 @@ defmodule SacaStatsWeb do
     end
   end
 
+  def generate_page_title(endpoint) do
+    site_name = "Saca Stats"
+    separator = " - "
+    postfix = separator <> site_name
+    endpoint_arr = String.split(endpoint, "/", trim: true)
+
+    case length(endpoint_arr) do
+      0 -> site_name
+      _ -> generate_subpage_title(endpoint_arr, separator) <> postfix
+    end
+  end
+
+  def generate_subpage_title([first | rest], separator)
+      when first == "character" and length([first | rest]) > 1,
+      do: generate_character_title([first | rest], separator)
+
+  def generate_subpage_title(endpoint_arr, separator) do
+    endpoint_arr
+    |> Enum.map_join(separator, fn endpoint -> String.capitalize(endpoint) end)
```

Comment:
It's good convention to not use pipes if there would only be one. You could also use the capture (`&`) syntax in place of an anonymous function
```suggestion
    Enum.map_join(endpoint_arr, separator, &String.capitalize(&1))
```

Reply:
Yeah I originally had multiple pipes, good catch